### DISCUSSION TITLE
Compress images on upload

### DISCUSSION THREAD

**Discussion by leandroperin** [ente/ente#555] [Enhancements]
I would like to see a feature that adds the option to auto compress images and videos on upload.

**Comment by alliedsincro** [comment]
Will this feature be added anytime soon? If not, then Ente can't be used for personal use as one takes several photos and videos during trips which one doesnt need in its full resolution when backed up to cloud.

If all those media gets uploaded at full resolution, then the storage quota will fill up exponentially faster than its competitors

**Comment by mathsen** [comment]
For me this is also a killer feature. If this would be available, it would be so great.

**Comment by gabriele2000** [comment]
Maybe let's add an internal converter with AVIF too and with quality levels (also add the lossless mode for AVIF so the user can fine-tune it)

**Reply by bxxcvrx** [reply]
+1

**Comment by iadaz** [comment]
Videos taken on a phone a always a lot larger than they need to be because phones don't/can't apply strong compression to them quickly enough. When I organise my videos I usually recompress them with ffmpeg and they usually come out under 30% of the original size with no appreciable loss of quality. Even a quick recompression might save quite a lot of space. If there's no capacity to do this in the cloud then an option to do this in the desktop app would be great. For me though, I'd still compress the originals to save space locally too.

**Reply by gabriele2000** [reply]
I'd propose to add a scheduled upload.
For example, the following conditions:
- When charging
- When the device isn't hot

Apply a compression (defined by the user, codec, quality level, quality preset) then upload them.

I think no one cares about an instant compression, but to not waste your data while uploading

**Reply by ahxxm** [reply]
Desktop might be the best place, the server is pretty lightweight and only keeps meta information, all encryption happens in client side, in which desktop is most performant and versatile

My preference is to only compress largest video/images, would be great if the app can sort by size, then select a few to compress in-place:
- video: simply `c:v libx265 -crf 24` should be enough for phone shoots and screencasts
- image: `cjpeg -quality 85`

(I do this by myself so it's fine for new contents, but existing ones are painful to compress again)

**Reply by iadaz** [reply]
> I'd propose to add a scheduled upload. For example, the following conditions:
> 
> * When charging
> * When the device isn't hot
> 
> Apply a compression (defined by the user, codec, quality level, quality preset) then upload them.
> 
> I think no one cares about an instant compression, but to not waste your data while uploading

I would also like it if the recompressed video had a custom meta tag that contained the compression settings used, a bit like https://superuser.com/a/1580036

**Reply by gabriele2000** [reply]
> Desktop might be the best place, the server is pretty lightweight and only keeps meta information, all encryption happens in client side, in which desktop is most performant and versatile
> 
> My preference is to only compress largest video/images, would be great if the app can sort by size, then select a few to compress in-place:
> 
>     * video: simply `c:v libx265 -crf 24` should be enough for phone shoots and screencasts
> 
>     * image: `cjpeg -quality 85`
> 
> 
> (I do this by myself so it's fine for new contents, but existing ones are painful to compress again)

Let users decide the codec and the quality rate factor.
Let users have a "pro" mode that can select everything and for "noobs" make them choose between "minimal quality loss" "medium quality loss" with examples generated on-the-fly or something like that.

I'm not gonna progressively lose details using ancient codecs (yes, jpeg is ancient by now) that degrades quality too much if you want to save space.

**Reply by ahxxm** [reply]
I had a different opinion on ancient and quality loss, for my fullframe DSLR images, `cjpeg -q85` strikes a balance between visually lossless and small file size, and I can share them to any os/device/platform without worrying about compatibility. 

As for candidate new formats I've tried, HEIC is not bad but even after these years it's not universally supported, newer formats like jpegli is more like a marginal improvement to mozjpeg, and jxl is yet to be adapted universally, I hope we can use jxl soon though, it's about 20%-25% smaller

**Reply by gabriele2000** [reply]
Key point: let the user decide.
Our opinion is irrilevant at the end of the day.

**Reply by ahxxm** [reply]
While 'letting the user decide' sounds 100% correct, the options we’re discussing require a level of background knowledge and understanding of the tradeoffs, for example obviously not all of users knows -q 85 and its corresponding butteraugli distance or visual loss, thus dumping options as a Pro mode could lead to poor decisions.

As a power user, I’d like to customize things like codec and CRF. However, it would be even better if there were well-thought-out presets that suit my needs before diving into Pro Mode—ideally, with links to reading materials for those who want to learn more.

**Comment by gabriele2000** [comment]
Any news devs?

**Comment by dhcgn** [comment]
A support of JPEG-XL (jxl) would be very helpful in the case of JPEG images. JPEG-XL is a modern image format that could address the compression need while offering additional benefits:

1. Space savings: JPEG-XL can **save around 20%** of space compared to standard JPEG **without losing any information**. It can also be losslessly transcoded back to the exact original JPEG if needed.

2. Improved quality: For new images, using JPEG-XL natively (not just transcoding) could provide even better compression and quality.

3. Additional features: JPEG-XL supports higher bit depths, wider color gamuts, and progressive decoding for faster perceived loading.

While JPEG-XL wouldn't address video compression, it could be a valuable part of an overall strategy to optimize storage use for images. Implementation would need to consider compatibility with various devices and platforms, but it could be a powerful option for users looking to maximize their storage efficiency.

See also: https://github.com/ente-io/ente/discussions/599 "Support uploading of JPEG XL"

**Reply by gabriele2000** [reply]
I'd rather have AVIF too, but I understand that would be "weird" because people see AVIF as a slow format, even if it can be fast as hell with the right encoding informations.

**Comment by stefanosweb** [comment]
Auto compress images and videos on upload would be great 👍

**Comment by handygaber2** [comment]
This would be a great addition to Ente!

**Comment by huntfx** [comment]
I take all my phone photos in HEIF at 9248x6936, so they work out around 15MB each. I have the files at full quality on my own PC, plus I run a backup, so I don't need Ente to triple up the storage. I just need a photo library to access at _reasonable_ quality on my phone.

Some options like these would go a long way without being too complex:

- Downscale all photos to [1500] x [1500]
- Resample all photos to [jpg quality 85]

I imagine pressing the button would:
1. Warn the user that it's not reversible
2. Do a search to find photos matching the criteria
3. Download each matching photo, process it, then upload it
4. Save the settings so any future photo uploads are processed automatically

I'd be more willing to pay for storage if I know it won't be eaten up by massive filesizes.

**Reply by gabriele2000** [reply]
I'd always let the user choose, for people who know what they're doing, like me (for example).
I'd like to use my encoding options without having a crappy downscaling or "jpeg to 85" because in my case I'd use ente as a backup service, since I have photos from 2009 too and I don't have a local backup.

**Reply by handygaber2** [reply]
I think the average user wouldn't want to think about compression. A non-aggressive form of compression would be awesome as a default (85 or 90). But you could always have a checkpox saying "enable high quality upload".

**Reply by iadaz** [reply]
I am using Ente Photos (instead of Amazon or Google) because I think/hope Ente will recognise that some users have the knowledge and expertise to configure the product how it best suits them.

- Sensible default for the masses 👍
- Flexible options for the pros 👍

**Reply by gabriele2000** [reply]
Exactly what @iadaz says.
@handygaber2 I understand that the average user wouldn't want to think about this "pro stuff", but let's at least avoid to gatekeep advanced options just because the average user wouldn't know how to use them.
Just put an "advanced" mode to let the user choose and that's it.

For what it's worth, here's my suggestions for different levels of compressions.

**Standard compression**
Codec: JPEG
Quality: 90%
Resolution: source (keep it the same)

**High compression** (with quality loss warning)
Codec: JPEG
Quality: 75%
Resolution: Limited but correctly scaled, setting only the height at 2048px; the width will be scaled with the proportion of the source resolution.

**Extreme compression** (with warning about support for older devices)
Codec: AVIF
Quality: 90%
Resolution: source (keep it the same)
If possible like on AV1 (AVIF is AV1 but with tuning for just encoding images), enable the "fast decode" flag, so the image won't take ages to decode on smartphones.

EDIT: I forgot to say: @handygaber2 "enable high quality upload" shouldn't be a checkbox.
The default should be to NOT TOUCH the source file, the compression should be manually activated (asked via a pop-up or  something) and should be doable even retroactively (meaning that if I have an entire uncompressed library I could choose to compress it using the settings that I'd like to use, triggering a recompression of everything).
A "keep the source files" would be cool too if someone wants to experiment with the various compression options.

**Reply by huntfx** [reply]
I wrote a quick Python script to convert a folder of images/videos, since I imagine the feature won't be added for quite some time.
It's currently a mess and I'll clean it up later, but if anyone else wants it, the link is [here](https://gist.github.com/huntfx/cc0891773fda1651b6cd6d45c8e27670).

- Converts images to max width/height of 2048 and saves as JPG quality 85, keeps EXIF data intact
- Converts videos to max width/height of 1280 with high ffmpeg compression

I ran it on a 63GB selection of photos/videos and the output was 5gb, so I should _just about_ be able to squeeze my 1TB library into 50GB if I skip the RAW files.

Edit: I've uploaded 19k videos/photos and it's used up 18GB, I've got _so much_ breathing room here.

**Reply by david-wagner-dev** [reply]
> I wrote a quick Python script to convert a folder of images/videos, since I imagine the feature won't be added for quite some time. It's currently a mess and I'll clean it up later, but if anyone else wants it, the link is [here](https://gist.github.com/huntfx/cc0891773fda1651b6cd6d45c8e27670).
> 
> * Converts images to max width/height of 2048 and saves as JPG quality 85, keeps EXIF data intact
> * Converts videos to max width/height of 1280 with high ffmpeg compression
> 
> I ran it on a 63GB selection of photos/videos and the output was 5gb, so I should _just about_ be able to squeeze my 1TB library into 50GB if I skip the RAW files.
> 
> Edit: I've uploaded 19k videos/photos and it's used up 18GB, I've got _so much_ breathing room here.

What would be a good workflow for this? For example if you take photos with mobile phone and then upload them then there would be no place for python script? This means you would need to upload them and then download them on desktop and run the script and then it syncs the changed photos?

**Reply by huntfx** [reply]
> What would be a good workflow for this? For example if you take photos with mobile phone and then upload them then there would be no place for python script? This means you would need to upload them and then download them on desktop and run the script and then it syncs the changed photos?

Yeah for this to work you'd need to have a copy of all your photos on the PC. It's unfortunately quite a manual process and certainly not for everyone. 

Think of it as an excuse to curate an offline version of your photo library.

1. I'll copy photos to PC and add/sort them to various albums in my main photo library. I note down which albums have changed.
2. Go to `cmd`, type `cd /path/to/photos`, and type `ente-convert /path/to/output`.
3. Assuming `path/to/output` contains existing files, it'll only process the new photos.
4. Drag and drop those changed albums back into Ente. It does a good job of de-duplication so will only upload the new photos.

Deleting photos is more of a pain as my script doesn't handle that, though as I type this I realise it would be easily possible, so I'll add the functionality soon.

**Reply by n3rdcub** [reply]
Should be possible via Termux if on android and _maybe_ iOS via a-shell

**Reply by huntfx** [reply]
I've updated the script by the way, it handles deletions now and puts everything in a subfolder automatically.

It would be great if Ente added a `post-upload` hook that would allow users to modify the images/videos, but I'd imagine something like that would be a nightmare to add security for and make cross platform.

**Comment by stefanosweb** [comment]
Since a lot of people use a phone it would be better to find an application for that. The most convenient way I am thinking is an auto compression app that has a specific output folder and then choose to auto upload that folder through ente.

Example: https://play.google.com/store/apps/details?id=com.ab.autoresizer

**Reply by gabriele2000** [reply]
No no and again no.
Autocompression app, with (maybe) stupid parameters that you can't choose?

**Comment by KeSch** [comment]
+1 for a compression option

Compression and/or resizing would be great, to have a choice if you want to save your photos in original or resized/compressed quality.

@enteio since one of your developers said in your subreddit already 3 years ago that this is on your roadmap, please let us know what your current implementation plan is.

**Reply by vishnukvmd** [reply]
We will look into this once there's a clear path towards writing the original metadata into compressed items during download / export across all clients.

**Reply by KeSch** [reply]
Thanks and keep my fingers crossed 🤞

**Comment by humble-new-student-89** [comment]
Same here, but I request jpeg xl over avif.  Reason is that the company and consortium behind avif (gooXX) is stopping lots of good things - for both avif and others (it booted out jpegxl from all its products - browser, os etc).

So in future, avif will not be as open as jpeg xl for any features - like gps exif metadata etc.

So, overall featureset wise and future proofing at same compression levels, jpeg xl is good.

**Reply by gabriele2000** [reply]
Everything would be solved by integrating an ffmpeg frontend and letting the user choose everything (with good defaults pre-set)

**Comment by kobus-v-schoor** [comment]
> We will look into this once there's a clear path towards writing the original metadata into compressed items during download / export across all clients.

@vishnukvmd FWIW, I use [exiftool](https://exiftool.org/) to transfer image and video metadata across after compressing my photos and videos. Haven't run into any issues and it preserves (almost) everything across formats.

+1 Would love to see this feature added, at the moment I manually download, re-compress and upload which is a pain.

**Comment by maddada** [comment]
+1 for this request. I always use the "Data Saver" setting on Google Photos and I won't be able to move to Ente unless photo/video compression is available. Otherwise storage will fill up MUCH faster compared to on Google Photos.
Please add this soon!

**Comment by szszl0** [comment]
Any news form Devs?

**Comment by n3rdcub** [comment]
Guys, what you’re requesting is a somewhat niche feature, useful sure but understand the requirements for such a thing involve back and front end work. 

You’ll need someone to update the hardware on the servers to accommodate the extra overhead (if not explore entirely new hosts to accommodate) as well as developers to bang about their keyboards to write efficient and clean code for this that what would normally be ran on a desktop to now be ran on a range of mobile devices. 

and this is hardly an exhaustive list, needless to say this isn’t something that will or would happen or be implemented in days, weeks, or even a couple of months. 

Let the devs cook. This request will take time

**Reply by gabriele2000** [reply]
Compression happens locally.
It doesn't take a mainframe to compress images.

As for the "required work by devs", we already have ffmpeg inside ente, as far as I remember seeing in a PR (that got merged months ago), and ffmpeg can do virtually everything.

**Comment by r3lativo** [comment]
+1 for a compression option

**Comment by laktosterror** [comment]
+1 for compression.

The compression could be done locally if on desktop. I'm mostly looking to compress big movie clips when using the freeing up space option in app.

**Reply by gabriele2000** [reply]
on phones too, with high "niceness" (low priority) in background, before upload

**Comment by MaxandreOgeret** [comment]
I really miss this feature

**Comment by laktosterror** [comment]
One cool feature would be if we could get smart filter based compression,  with use of already existing ML. For example to not compress pictures of landscapes and people

**Comment by shompoe** [comment]
I just installed Ente to try it out and I was quite disappointed that photos are uploaded at their full resolution only. You could buy a sizeable subscription and just fill it up very quickly with data you simply do not need nor wish to keep.

**Comment by jorge2000x** [comment]
It's a missing feature; all that's needed is to add an opt-in option and use a newer algorithm, JXL is the most suitable. Nowadays we still use the old JPEG; I've been doing some conversions on my cold backups, converting from JPEG to AVIF (which is more efficient but I don't think it's ideal) reduced the size of most images from 3-5MB to... 250-500KB, Without losing much quality, except for camera noise (JPEGXL manages to preserve noise a little better). It's a huge difference in size; you could upload many more photos in the same space without losing much quality. Remember, this should be an opt-in option.

**Reply by gabriele2000** [reply]
I actually converted all of them to AVIF when transferring them from Google Photos, using a CQ of 18, which preserves everything visually, but in size there's a HUGE reduction.
We're talking about stuff that retains everything, except a 2% of quality maybe... it's an acceptable loss or a 80% of size reduction, even more depending on the case.

**Comment by chinkung** [comment]
It is 2026, any update on this feature.

**Comment by Grosbof** [comment]
Same asking over here. New phones have larger and larger sensors where, even with some configuration on the camera app, files sizes are larger and larger. Compression solve this problem while downsizing the photos with a perfectly and almost unnoticeable loss of quality.

**Comment by EndrII** [comment]
I also noticed that there's an archive feature, and it could be improved. Essentially, users could upload their old photos and videos to the archive—or, for example, have them uploaded after, say, a month (automatically)—and everything that ends up in the archive would be compressed to a readable level sufficient for memories. For example, videos up to 720p or full HD (without fps loss) and photos up to 2K.
And, of course, it would be possible to choose the compression level.

**Reply by gabriele2000** [reply]
To archive something means to store it in the best futureproof format, which means FFV1 for videos (lossless) and PNG for images (lossless).

So no, let's not do it in that section please.

**Comment by r3lativo** [comment]
I was looking for a separated solution and I found out FFShare: however as the name suggests it only compresses one file at a time that is sent through the app via sharing feature.

Looking for a way to compress locally, I was able to set up a routine with Termux, running a bash script to compress videos and images and sending them to a specific folder. I was able to compress 5.5GB of files to only 2.2GB (images to 85% quality, videos set to 28 on ffmpeg compression), while also having the option to remove Exif data and GPS data. Ente also offers upload via CLI, but i didn't try it, neither via computer nor phone.

This approach is not optimal, but it leaves room for a better solution. I expect that locally compressing right away might create problems, so a nightly local routine could work?

**Comment by notAunty** [comment]
Same here, would like to have an offline optimized copy kept on device when I use the "Free up storage" option, just like the iCloud Photos integration on iPhone Photos app. When user revisit the (locally) deleted photos, and try to zoom in on it, the full-sized image would then be loaded.

I've created an initial code merge request here: https://github.com/ente-io/ente/discussions/9190

**Reply by EndrII** [reply]
You're proposing a feature that saves space on your device, not in cloud storage. The idea here is to send files compressed to a certain value to the cloud, rather than the original quality, to save space. The idea is that if you've been storing photos and videos for decades, you don't need 4K resolution, even though you usually do immediately after shooting. Therefore, the app should be able to compress old videos and photos to acceptable values, allowing you to store as many photos as possible in your cloud. Currently, only 2 TB of storage is available—a rather modest amount, considering the size of photos and videos in 2026.

**Reply by Grosbof** [reply]
Spot on Endrll

**Reply by notAunty** [reply]
I see, thanks for clarifying. Both features is important, in my opinion, given 4k HDR10+ or even 8k videos are coming soon.

**Reply by EndrII** [reply]
@notAunty  Sounds like the Ente Photos already have similar function that you described
see https://ente.io/help/photos/faq/backup-and-sync#optimize-iphone-storage

**Reply by notAunty** [reply]
@EndrII 
That's for people who use iCloud photos. I don't think many have both subscriptions together.

What I'm describing replicates the iCloud Photo feature, where instead of deleting local copies of backed up photos outright, it keeps an optimized version on iOS, Android (Google photos don't have this feature) and others.

In other words, this just brings iCloud photos feature parity to ente-only users on iOS. And improve the Android experience.

PS. Given the long standing of this Discussion, I don't think what I'm describing would see the light of day. I might just fork the project, and implement it myself client-side.

**Reply by r3lativo** [reply]
I'll share here some of the difficulties I found when I tried to do it locally.

Retaining HDR and motion photos after compression was not feasible for me.

Another thing to keep in mind is (for Android) folder permissions, i.e. in my case saving images to a folder which is not DCIM/Camera inhibits the system to also maintain the actual time and date, and an Ente backup would result in photos with backup time as the associated time. This does not always happen, I had no problem with another smartphone.

**Comment by gordielachance** [comment]
Same here.  That would really make the difference.
I need a cloud space to preview my photos, not to do a full high resolution backup (+ I can't afford it).

### My ideal uploader

1. define a **watch folder** (Ente does already this) 
2. add rules to declare **which files/folders should be ignored**  (maybe through a _.enteignore_ file, just like a _.gitignore_ does for git)
3. define **preprocess settings** for both images and videos (maximum resolution, maximum compression).  It should also copy the metadatas of the original files (ex. to keep the initial video creation date)

It would seems logic that the preprocessor would create a _processed_ subfolder within the watched folder; and that the real sync would be done based on that subfolder.

Then I could finally get rid of Google Photos and not having to code my own tool.

Thanks a lot.

**Reply by EndrII** [reply]
It's best to avoid creating duplicates. Store the hash or ID of the original in the compressed video or photo, so that the compressed version is stored in the cloud and the original is stored on the device. Duplicates will cause confusion or inconsistency, and they also open the door to logical bugs.

**Reply by iadaz** [reply]
Regarding being able to ignore certain files/folders (gitignore style) I created a feature request for this here: https://github.com/ente-io/ente/discussions/2878

**Comment by manquetin** [comment]
+1 for compression option

**Comment by boesemar** [comment]
+1 Really needed feature to be able to move from google photos! This saves a lot of storage!

**Comment by DavinKaru** [comment]
Bump. This is the deal breaker that's keeping me from swapping from Google Photos to Ente!

I take a lot of pictures, but I very rarely need something in full resolution (if I did, they'd be stored on Drive, not photos gallery)..

**Comment by PedroC88** [comment]
Please!

**Reply by PedroC88** [reply]
I installed a self hosted instance to test it and only uploaded my most recent photos that I already had on my phone, that took 11GB out of my self-assigned 100Gb. At the original sizes it would be unsustainable to migrate from Google Photos (which is what I want to do).

**Comment by SourLemonJuice** [comment]
I'm using a Google Pixel phone, the video feature of its camera will always record in 20+ mbps bitrate. A one minute video costs 150MB of storage on my phone.
I know it's not the Ente's fault, but I'm still unclear on which workflow is the best for video backups. As for the Ente, it is hard to copy my existing videos onto the 50+10GB cloud :[

**Reply by PedroC88** [reply]
Yeah, I have the same issue. I looked at the settings on my phone and, even on the lowest settings, media is just too large to sustain ente uncompressed as my permanent backup.

I really hope ente comes up with a solution.

**Comment by PedroC88** [comment]
If Ente is reading this, if you guys can provide some architectural guidance on how this should/would be implemented, I can attempt to do it, I just don’t know enough about the code structure to try this without some guidance.

**Reply by PedroC88** [reply]
Here's a sketch I put together quickly to get this conversation started if possible (also a mock up of what the screen might look like).

The idea is that the client would generate a transcoded/compressed **copy** of the media and then upload that to the server. The server should accept two hashes for the original and transcoded files just for validation and internal logic but, if the user is pushing transcoded media, only that copy would be uploaded to the storage, effectively losing the original once it's removed from the device (this could be revisited in the future to maybe keep both files, but this would a least make this work as Google Photos).

<img width="562" height="487" alt="page-1" src="https://github.com/user-attachments/assets/7f440790-c8cb-499b-a5cb-a7feb1cc5276" />
<img width="222" height="482" alt="page-2" src="https://github.com/user-attachments/assets/4f2c67f8-a86b-4fca-a65a-556ad3d82f58" />

**Reply by gordielachance** [reply]
should also keep the original metadatas (ex. video creation date) so the items keep being sorted correctly.

**Comment by gordielachance** [comment]
Hi, I won't renew my subscription unless I see this feature. Sorry, not handy enough currently.

**Comment by enteio** [comment]
Hi, 

Thank you for the feedback. I have shared it with our dev team and they have taken it into consideration.

If there is anything else you need help with, please reach out.

Best,
Pratty






If you like ente, please consider leaving a review on https://play.google.com/store/apps/details?id=io.ente.photos , https://apps.apple.com/us/app/ente-photos/id1542026904  or https://alternativeto.net/software/ente/about/ .










---- on Wed, 13 May 2026 21:27:35 +0530  "Pedro ***@***.***>  wrote ---- 



Here's a sketch I put together quickly to get this conversation started if possible (also a mock up of what the screen might look like).
The idea is that the client would generate a transcoded/compressed copy of the media and then upload that to the server. The server should accept two hashes for the original and transcoded files just for validation and internal logic but, if the user is pushing transcoded media, only that copy would be uploaded to the storage, effectively losing the original once it's removed from the device (this could be revisited in the future to maybe keep both files, but this would a least make this work as Google Photos).
https://github.com/user-attachments/assets/7f440790-c8cb-499b-a5cb-a7feb1cc5276 
https://github.com/user-attachments/assets/4f2c67f8-a86b-4fca-a65a-556ad3d82f58 


—
Reply to this email directly, https://github.com/ente-io/ente/discussions/555#discussioncomment-16907177 , or https://github.com/notifications/unsubscribe-auth/A4HZ5FQDUILGB7Z5R3A5VUL42SLO7AVCNFSM6AAAAABONIAIT2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTMOJQG4YTONY .
Triage notifications on the go with GitHub Mobile for https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675  or https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub .

**Reply by DavinKaru** [reply]
Hi, is there any update on this feature? With Google Photos' Personal Intelligence having rolled out I want to be able to swap but isn't feasible without this feature.

**Reply by leandroperin** [reply]
@enteio we're almost 3 years waiting for a position here. it's about time to tell us if this feature will be built or not

**Comment by MaxandreOgeret** [comment]
@enteio I suspect this feature will never come. This is not in your interest to compress images on upload because users would not have to pay for as much storage.

**Reply by gordielachance** [reply]
...and that's a mistake, because by offering a feature that many people are looking for, you could attract users who have so far been too hesitant to subscribe—and/or retain subscribers who are going to leave (like me) because that feature isn't coming. ---> Gain customers, spare ressources.

**Reply by EndrII** [reply]
and also, I will not renew the subscription, because why should I do this if it is possible to use Google Photos at half the price, and with this feature. 
So this year is ending, and my terabyte is migrating to Google again.

So let them continue implementing useless functions in the form of memories remade for the 10th time, I really don’t like it when the wrong priorities are chosen.

**Reply by EndrII** [reply]
and I  want to say that among open source, you are the best, Ente. But if crowd of people are asking for something, you have to listen it

**Reply by JulioHC00** [reply]
Will add that I've considered Ente before but didn't choose them because of the lack of this feature.

**Reply by adangert** [reply]
Looking into this as a first time user (Also using Google drive), this is the one feature that is not supported, that's very important, the compression on google drive is very good (basically like the original and saves tons of space), I wish it was here, will be sticking with Google Drive unfortunately for now. 😞

**Comment by chiraag-nataraj** [comment]
Perhaps I'm missing something, but I honestly don't see the point of a feature like this. It feels like there are two dominant workflows and I don't see how (lossy) compression is desirable in either case.

1. You're using Ente as a backup of all of your photos. You have autobackup enabled and everything gets automatically uploaded to Ente. In this case, lossy compression does not make sense because at that point it's no longer a backup. By definition, a backup should simply be a copy --- no modification other than encryption and perhaps lossless encryption. You should be able to recover the original file from the version uploaded to Ente.
2. You're using Ente as a way to share selected photos with others. You selectively upload photos (either processed RAW files turned into JPEGs or specific photos from your phone). In this case, the version on Ente is recoverable from local data (you can re-export the pictures or re-upload the specific ones from your phone), so Ente is no longer considered a backup (that happens elsewhere in the manner described in workflow 1, but maybe to Proton Drive or Google Drive or whatever). In this case, compression may perhaps help, but only in the case that you have tons of phone photos you want to upload and you have no other way of pre-compressing them (in the RAW case, you can set compression settings during export anyway).

People seem to be clamoring for compression within the context of workflow 1, but that genuinely makes no sense. If you consider something as a backup, you should be able to recover the exact original file from the backup --- otherwise it's not a backup.

My workflow is mostly workflow 2 on a self-hosted Ente (storage backend is Backblaze B2). Most photos are either coming from phones (I can deal with compressing them later if necessary) or RAW images (already mostly optimized for what I want).

I would honestly argue that for workflow 1, you're probably better off self-hosting and using an object storage backend to keep costs reasonable (I'm currently paying $7/TB/mo and I'm currently using 5 GB in my self-hosted Ente, but that's for workflow 2). Because of the client-side encryption, you don't have to particularly trust your object storage provider either...just make sure they're reliable and have multiple backups!

**Reply by teohhanhui** [reply]
You have chosen to see the technical side of things, but overlooked the social aspect. A "backup" in terms of photos and videos doesn't necessarily mean a lossless copy, except for a very specific niche of people who want to keep all of their digital life pristine. That is simply *not* what the general public (or even many power users) wants. Photo / video backup is largely about preserving "memories".

**Reply by chiraag-nataraj** [reply]
> A "backup" in terms of photos and videos doesn't necessarily mean a lossless copy, except for a very specific niche of people who want to keep all of their digital life pristine.

IDK, the main reason I chose Ente when looking for ways to self-host an image gallery was that it _is_ lossless. And there are lots of people like me who dislike the fact that something like Google Photos isn't necessarily lossless (I'm not even sure what they do anymore). Drive or object storage is fine for lossless storage, but gets rid of the easy gallery and sharing aspect (the other main thing I was looking for). I have _never_ known a 'backup' to not be a pristine copy --- like, just in terminology, a 'backup' that lossily compresses your files is just using the wrong term. It doesn't mean it isn't useful --- for example, when sharing photos with others (or on the Internet), maybe you don't really care about whether it's compressed or not. But we shouldn't dilute the term 'backup' to mean any copy of a file (it's already bad enough that people consider cloud storage to be a backup of their files).

> That is simply not what the general public (or even many power users) wants. Photo / video backup is largely about preserving "memories".

Yeah, it absolutely it is about preserving memories. The thing is, though...what does 'preservation' mean here? Does it just mean _a_ copy? If you copy an old camcorder tape (we've done a bunch of that) and there's static and parts of the video are gone, is that enough to say the memory is preserved? If the digital version is significantly worse quality than the analog version, is that sufficiently preserved? These are hard questions, to be honest, and there are no right answers. Like, I'm not saying any of this to be snarky, these really _are_ tough questions, especially when talking about analog to digital conversion.

Regarding compression, I don't necessarily mind if it's an option, but I would be really scared to have it be the default. Ente's huge promise (written all over the place, including in the docs) is that they don't alter your files. Introducing lossy compression by default would suddenly change behavior for existing customers and self-hosters (and would make it really easy to accidentally store lesser quality than intended).

There are enough of us already annoyed that editing photos within Ente wipes the metadata (and I'm sure many people have lost files because they thought the editing preserved the metadata when it didn't). We *already* have failure modes that are unacceptable (that they are presumably working on!). I don't think introducing yet another failure mode, yet another way to store a different version of your data than you intended, is a great idea.

Anyway, clearly I touched a bunch of nerves with my viewpoint. Is it a dealbreaker if Ente introduces lossy compression? Not really (assuming it's just an option I can turn off). But would it make me scared that there will be unintended consequences and people will accidentally store lower-quality versions than they intended? Absolutely.

Whichever side of this you're on, I think taking into account the fact that the default until now **has been lossless uploads** is important. Suddenly allowing for that behavior to change, even just as an option, will have unintended effects.

**Reply by gordielachance** [reply]
i'm more in workflow 2:

_...but only in the case that you have tons of phone photos_

**of course** we have tons of photos!  

1. That's why we need a way to "view" them (which is what Ente does); 
2. That's why compression should be possible to avoid useless high prices; 
3. That's why it should be an Ente feature rather than letting everyone searching/building their own solution.

(btw, most of them are not phone photos - might from real cameras or analogic photos scans -which can be waaayyy heavier)

**Reply by PedroC88** [reply]
IMO, if you don’t care about this feature then just ignore this thread and continue about your day.

I find the direction of this conversation unproductive and I see no point. Some of us would like to see it which is why this is one of the most upvoted threads (whether Ente cares or not), for those that don’t care then good for you.

Or is it your point to convince us that we are wrong in wanting this and we should thank you for enlightening us that we don’t really need this? Call those “backups” whatever you want but I can’t afford the storage to “save” (to use your terms) my images in uncompressed sizes.

**Reply by gabriele2000** [reply]
@chiraag-nataraj I am a power user.
I love to take photos in the highest quality, though I can have an almost lossless quality using AVIF at CRF 15, while using 50% less data than a JPEG at 100% quality and 5% data than a DNG file.

Make of this what you want, personally if I lose 5% of perceptual quality for 500% of space saving, so be it.
It's the reason I encode videos in AV1 to share with people, also because space is precious and network is precious for some people, ain't not way I am going to waste space and time.

For music I use FLAC, yet if I want to share something with people I will re-convert it to OPUS 320kbps, it's near-lossless and it weights like... 5 times less?

So yeah, usually my photos are original quality in my microSD inside my phone, or inside the phone's internal storage, I backup to have everything saved, physical backups are to not pay €200000 in storage online, it's really the worst case scenario to lose everything and rely on my cloud backup.

ALSO take my scenario: family shared account, my dad has so many photos taken from his phone, every photo weights 5/6MB, he's occupying 40GB alone, while I am just using 20GB and I have 35000 photos, 80% of them compressed in AVIF while being imported from google photo cloud.

What if I want to compress his photos in AVIF with quality factor 15 with a button?

**Reply by DavinKaru** [reply]
@chiraag-nataraj, the post this Discussion was opened with, by @leandroperin, reads "...adds the option". To be absolutely clear, the **option** for other users, those interested and who have a use-case for it, to compress images on upload **does not forbid you from uploading losslessly.**

**Reply by gabriele2000** [reply]
> @chiraag-nataraj, the post this Discussion was opened with, by @leandroperin, reads "...adds the option". To be absolutely clear, the **option** for other users, those interested and who have a use-case for it, to compress images on upload **does not forbid you from uploading losslessly.**

To further add, a while ago I had the idea to use ffmpeg, to be able to do this locally, maybe even in the background.
For ente this wouldn't be a cost, and it would also respect the "local-first" policy, like it's with AI.

**Reply by adangert** [reply]
To note: in Google photos, I only enabled compression for photo backups after doing research, and finding that it's basically impossible to notice differences as a human in detail when the correct compression algorithm is chosen, while saving something like 75% of storage space. Basically was a no brainer at that point, and I'm glad Google seemingly did research on this!

An example article on this: https://vatthikorn.com/google-photos-compression/

**Reply by chiraag-nataraj** [reply]
Clearly I'm in the minority here :) I'll peace out of this now haha. I tend to be in the camp of wanting full control over everything, so one of the reasons I selected Ente was _because_ it didn't edit my files in any way. Clearly others feel differently, which is fine.

I also use FLAC and export my processed RAW files as JPG at 90% quality, so it's not like I'm _opposed_ to doing such things. It's just that I want full control over the process :) Anyway, just one user's opinion (and clearly in the minority at that!). As long as it remains an option (and hopefully off by default), it's obviously fine (since I'll just turn it off if it's enabled on an update or whatever).

**Reply by bordam** [reply]
Should definitely be off by default since it’s an irreversible action that is destructive to some extent

**Comment by adrienbar** [comment]
In addition, regardless of the compression ratio, transcoding to format natively supported by clients (Desktop electron app, ios, android) would prevent lag and uneeded battery-consumption in the viewing experience due to live and systematic client side transcoding.
For instance heic is not supported by chromium (electron) which forces the client app the transcode to jpeg on the fly on every view. If you navigate fast in your gallery it becames laggy and CPU is heating :-( 

Upstream avih or wepb transcoding would make the experience smoother :-)
